modules/packages/OrderedMap.chpl

```diff
@@ -0,0 +1,746 @@
+      guaranteed to be in any particular order.
+
+      :return: A new DefaultRectangular array.
+      :rtype: [] valType
+    */
+    proc valuesToArray(): [] valType {
+      _enter(); defer _leave();
+
+      if !isCopyableType(valType) then
+        compilerError("valuesToArray requires a copyable value type");
+
+      var A: [0..#size] valType;
+      for (a, v) in zip(A, values()) {
+        a = v;
+      }
+      return A;
+    }
+  } // end record orderedMap
+
+  /*
+    Replace the content of this orderedMap with the other's.
+
+    .. warning::
+
+      This will invalidate any references to elements previously contained in
+      `lhs`.
+
+    :arg lhs: The orderedMap to assign to.
+    :arg rhs: The orderedMap to assign from. 
+  */
+  proc =(ref lhs: orderedMap(?kt, ?vt, ?ps), const ref rhs: orderedMap(kt, vt, ps)) {
+
+    if !isCopyableType(kt) || !isCopyableType(vt) then
+      compilerError("assigning orderedMap with non-copyable type");
+
+    lhs.clear();
+    for key in rhs.keys() {
+      lhs.add(key, rhs[key]);
+    }
+  }
+
+
+  /*
+    Returns `true` if the contents of two orderedMaps are the same.
+
+    :arg a: A orderedMap to compare.
+    :type a: orderedMap
+
+    :arg b: A orderedMap to compare.
+    :type b: orderedMap (with same keyType and valType)
+
+    :return: `true` if the contents of two orderedMaps are equal.
+    :rtype: `bool`
+  */
+  proc ==(const ref a: orderedMap(?kt, ?vt, ?ps), const ref b: orderedMap(kt, vt, ps)): bool {
+    if a.size != b.size then return false;
+    for (e1, e2) in zip(a.items(), b.items()) {
+      if e1 != e2 then return false;
+    } 
+    return true;
+  }
+
+  /*
+    Returns `true` if the contents of two orderedMaps are not the same.
+
+    :arg a: A orderedMap to compare.
+    :type a: orderedMap
+
+    :arg b: A orderedMap to compare.
+    :type b: orderedMap (with same keyType and valType)
+
+    :return: `true` if the contents of two orderedMaps are not equal.
+    :rtype: `bool`
+  */
+  proc !=(const ref a: orderedMap(?kt, ?vt, ?ps), const ref b: orderedMap(kt, vt, ps)): bool {
+    return !(a == b);
+  }
+
+  /*
+    Returns a new orderedMap containing the keys and values in either a or b.
+  */
+  proc +(a: orderedMap(?keyType, ?valueType, ?parSafe),
+         b: orderedMap(keyType, valueType, parSafe)) {
+    return a | b;
+  }
+
+  /*
+    Sets the left-hand side orderedMap to contain the keys and values in either
+    a or b.
+   */
+  proc +=(ref a: orderedMap(?keyType, ?valueType, ?parSafe),
+          b: orderedMap(keyType, valueType, parSafe)) {
+    a |= b;
+  }
+
+  /*
+    Returns a new orderedMap containing the keys and values in either a or b.
+  */
+  proc |(a: orderedMap(?keyType, ?valueType, ?parSafe),
+         b: orderedMap(keyType, valueType, parSafe)) {
+    var newMap = new orderedMap(keyType, valueType, parSafe, a.comparator);
+
+    for e in a.items() do newMap.add(e[0], e[1]);
+    for e in b.items() do newMap.add(e[0], e[1]);
```

Comment:
Since you've written `|=` below, you can write this as (the equivalent but cleaner IMO)

```chapel
newMap |= a;
newMap |= b;
```